crates/bins/src/bin/datadog-static-analyzer.rs

```diff
@@ -549,141 +571,127 @@ fn main() -> Result<()> {
     ////////////////////////////////////////////////////////////////////////////////////////
     // Secrets Test
     #[allow(unused_mut)]
-    let mut detected_secrets = Vec::<DetectedSecret>::new();
+    let mut detected_secrets = Vec::<SecretResult>::new();
     #[allow(unused_mut)]
     let mut secrets_rules = Vec::<SecretRule>::new();
     #[cfg(feature = "secrets")]
-    if matches.opt_present("test-secrets") {
-        use cli::secrets::ValidationStatus;
-        use secrets::temp_integration_test::{build_secrets_engine, shannon_entropy};
+    if scan_for_secrets && secrets_rule_file.is_some() {
+        use cli::secrets::{as_position, ValidationStatus};
+        use secrets::core::validator::Candidate;
+        use secrets::ScannerBuilder;
         use std::sync::mpsc;
-        use std::sync::mpsc::RecvTimeoutError;
         use std::sync::Arc;
         use std::time::Duration;
+        use threadpool::ThreadPool;
```

Comment:
Why using this? I believe we said everything happened sequentially and we will rely on having `rayon` doing all parallelism?